### PR TITLE
[NetBeans-1019]False positive of "Null Pointer Dereference" when passing a lambda with a null possible result to a non-null argument

### PR DESCRIPTION
**Pull request by vikasprabhakar**
Jira Id : https://issues.apache.org/jira/browse/NETBEANS-1019

Issues: False positive of "Null Pointer Dereference" when passing a lambda with a null possible result to a non-null argument
private void func(@NonNull Function<Object, Object> func){
        
    }
    @CheckForNull
    private Object foo(){
        return null;
    }
    private void bar(){
        func(item->foo());//false positive
    }

### CODE REVIEW

File: java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java [hunk 1]
```diff
@@ -625,7 +626,7 @@ public State scan(Tree tree, Void p) {
             
             TypeMirror currentType = tree != null ? info.getTrees().getTypeMirror(new TreePath(getCurrentPath(), tree)) : null;
             
-            if (currentType != null && currentType.getKind().isPrimitive()) {
+            if ((tree != null && tree instanceof JCTree.JCLambda) || (currentType != null && currentType.getKind().isPrimitive())) {
```
Comment (rtaneja1) [inline java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java]: You can use tree.getKind in place of instanceof

Comment (sdedic) [reply java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java]: in fact `getKind` is strongly preferred, as it allows the implementor to mix in interfaces. Please void dependencies on `JC*` implementation classes from Javac, if possible.

Comment (vikasprabhakar) [reply java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java]: getKind() method added.

File: java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java [hunk 2]
```diff
@@ -626,7 +625,7 @@ public State scan(Tree tree, Void p) {
             
             TypeMirror currentType = tree != null ? info.getTrees().getTypeMirror(new TreePath(getCurrentPath(), tree)) : null;
             
-            if ((tree != null && tree instanceof JCTree.JCLambda) || (currentType != null && currentType.getKind().isPrimitive())) {
+            if ((tree != null && tree.getKind().equals(Kind.LAMBDA_EXPRESSION)) || (currentType != null && currentType.getKind().isPrimitive())) {
```
Comment (sdedic) [inline java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java]: `Kind.LAMBDA_EXPRESSION` is an num, can be compared using `== `

**Pull request merged**